src/Template/Loggers/IOHprofiler_ecdf_logger.h

```diff
@@ -194,8 +195,19 @@ class IOHprofiler_ecdf_logger : public IOHprofiler_observer<T>
          * First index: problem id,
          * second index: instance id,
          * third index: dimension id.
+         * fourth index: run id
          */
-        const IOHprofiler_AttainMat& at(size_t problem_id, size_t instance_id, size_t dim_id) const;
+        const IOHprofiler_AttainMat& at(size_t problem_id, size_t instance_id, size_t dim_id, size_t run_id) const;
+
+        /** Access an averaged attainment matrix of multiple runs on a problem
+         *
+         * @note Use the same indices order than IOHprofiler_problem.
+         *
+         * First index: problem id,
+         * second index: instance id,
+         * third index: dimension id.
+         */
+        IOHprofiler_AttainMat at(size_t problem_id, size_t instance_id, size_t dim_id) const;
```

Comment:
This pushes assumptions on how the data-structure built by this class will be used. I would find more logical to use an `ecdf_stat` sub-classe to compute such a threshold matrix.

Reply:
Indeed, and the 0.5 constant also produces confusion. I have removed this function from the ecdf_logger, will create another class for other functionalities.